sidecar/supervisor/init.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2016 IBM Corporation
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package supervisor
+
+import (
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+
+	"github.com/Sirupsen/logrus"
+)
+
+// Init launches new sidecar process with same args and env vars as current process
+// and runs zombie process cleanup for any inherited orphans
+func Init() {
+	cmd := exec.Command("a8sidecar", os.Args[1:]...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = os.Environ()
```

Comment:
No need to explicitly copy the environment variables, as by default they are inherited from the parent.